src/backend/expungeservice/expungement_analyzer/expunger.py

```diff
@@ -0,0 +1,89 @@
+from expungeservice.expungement_analyzer.analyzers.type_analyzer import TypeAnalyzer
+from expungeservice.expungement_analyzer.analyzers.time_analyzer import TimeAnalyzer
+
+
+class Expunger:
+    """
+    This is more or less a wrapper for the time_analyzer and type_analyzer.
+    After running this method the results can be extracted from the cases
+    attribute. The errors attribute will list the reasons why the run
+    method failed to evaluate in which case the run method will return
+    False; otherwise there were no errors and it returns True.
+
+    Most of the algorithms in this class can be replaced with database
+    query's if/when we start persisting the model objects to the db.
+    """
+
+    def __init__(self, cases):
+        '''
+        Constructor
+
+        :param cases: A list of cases
+        '''
+        self.cases = cases
+        self.errors = []
+        self._charges = []
+        self._most_recent_dismissal = None
+        self._most_recent_conviction = None
+        self._second_most_recent_conviction = None
+        self._num_acquittals = 0
+        self._acquittals = []
+        self._convictions = []
+        self._time_analyzer = None
+
+    def run(self):
+        """
+        Evaluates the expungement eligibility of a record.
+
+        :return: True if there are no open cases; otherwise False
+        """
+        if self._open_cases():
+            self.errors.append('Open cases exist')
+            return False
+
+        self._create_charge_list()
+        self._categorize_charges()
+        self._set_most_recent_dismissal()
+        self._set_most_recent_convictions()
+        self._set_num_acquittals()
+        TypeAnalyzer.evaluate(self._charges)
+        self._time_analyzer = TimeAnalyzer(most_recent_conviction=self._most_recent_conviction,
+                                           second_most_recent_conviction=self._second_most_recent_conviction,
+                                           most_recent_dismissal=self._most_recent_dismissal,
+                                           num_acquittals=self._num_acquittals)
+        self._time_analyzer.evaluate(self._charges)
+        return True
+
+    def _open_cases(self):
+        for case in self.cases:
+            if case.current_status != 'Closed':
+                return True
+        return False
+
+    def _create_charge_list(self):
+        for case in self.cases:
+            self._charges.extend(case.charges)
+
+    def _categorize_charges(self):
+        for charge in self._charges:
+            if charge.acquitted():
+                self._acquittals.append(charge)
+            else:
+                self._convictions.append(charge)
+
+    def _set_most_recent_dismissal(self):
+        self._acquittals.sort(key=lambda charge: charge.date)
+        if self._acquittals and self._acquittals[-1].recent_acquittal():
```

Comment:
Shouldn't this be len(self.acquittals)?

Reply:
Can you expand on your question. Not sure what your asking here.

Reply:
Yes, line 76. I was saying the check should be `if len(self._acquittals) and ...` or `if len(self._acquittals) > 0 and ...` like line 81. If `self._acquittals` is an empty list, `self._acquittals[-1]` would throw an exception.

Reply:
No, it doesn't throw an exception if `self._acquittals` is an empty list. An empty lists truth value is false. https://docs.python.org/3/library/stdtypes.html#truth-value-testing

Reply:
and so it would short circuit and not evaluate `self._acquittals[-1]`

Reply:
Cool. Then we are good here.